modules/engage-paella-player/src/main/paella-opencast/plugins/es.upv.paella.opencast.loader/03_oc_search_converter.js

```diff
@@ -215,11 +215,17 @@ class OpencastToPaellaConverter {
         return (((smask[0] == '*') || (smask[0] == sflavour[0])) && ((smask[1] == '*') || (smask[1] == sflavour[1])));
       });
 
-      let importT = filterStream.tracks.tags.some(function(cTag) {
-        return currentTrack.tags.tag.some(function(t){
-          return ((cTag == '*') || (cTag == t));
+      let importT = false;
+      if (currentTrack.tags.tag) {
+        if (!(currentTrack.tags.tag instanceof Array)) {
+          currentTrack.tags.tag = [currentTrack.tags.tag];
+        }
+        importT = filterStream.tracks.tags.some(function(cTag) {
+          return currentTrack.tags.tag.some(function(t){
+            return ((cTag == '*') || (cTag == t));
+          });
         });
```

Comment:
@miesgre This solves the problem is an Opencast mediapackage TRACK element has no tags such as to "archive" the track or to publish it to "engage".... if the track was never tagged to be published to the search index. Where is Paella getting it from? I suspect this is related to the other recent "bug fix"  issue where error handing is needed for published events which have no creation date nor even title.

Reply:
I think, I see where the confusion comes from: @karendolan, you don't need to tag a track to publish to engage. The `publish-engage` operation has always supported publication by flavor only. Tagging tracks is just a convenient way to mark things for publication without having to remember later on what needs publication.

Reply:
@karendolan Paella is getting the information from the search service. As @lkiesow there is no need to tag a track to upload to the engage. See the `download-source-flavors` key of the `publish-engage` operation.  But I found this problem on live events. When there is a live events, the tracks haven't tags.